### ISSUE TITLE
Broken IPFS File?

### ISSUE THREAD

**Issue by lautarodragan** [poetapp/node#53]
This file is split in half, which means it's invalid JSON and will be rejected with Po.et Nodes.

https://ipfs.io/ipfs/QmTjDerwKUXDVosZ7nhYXSpsn4arFKfwXVcQuqxh3N7Hin

The article of origin seems to be this one: https://uuotv.com/2018/02/14/bcx-2/.

Most likely came from Frost.

- If the node received it broken in the first place - why didn't it get rejected immediately?
- If the node received it correctly - why is the IPFS file broken?

Possibly an issue with the request size when POSTing to IPFS, or a configuration issue with IPFS.

https://github.com/poetapp/node/blob/aad2f4804de85646875257158d1ea465a03af030/src/Storage/IPFS.ts#L26-L44

Update:

[This file](https://ipfs.io/ipfs/QmTirShRAVTdHWyj8u2tXmNoNv4ggXZJLkZoo3rvZPtJhi) is about 10 times the size and isn't broken. Maybe an issue with a particular character, or language / encoding?

Update2:

There's files in plain english that are broken too, so unlikely to be an encoding / language issue. 

Needs #119

**Comment by lautarodragan** [comment]
@warrenv @wzalazar this can be super tricky and hard to reproduce, but we probably want to be aware of it, at the very least.

The only way that this should be possible is if people injected data into the Po.et protocol manually or with a third party application instead of using the Po.et node or Frost, but it's very unlikely someone is doing that...

**Comment by geoffturk** [comment]
Setup separate testing issue (func. tests) - check if nodes sync well (looks up hash in IPFS), check hash matches original file

**Comment by geoffturk** [comment]
Here's another broken one I found randomly while testing out node today:
https://ipfs.io/ipfs/Qma7Kin2F4Cx88DS7XFy8gs9vY2uZWspw2jnx36fTURjF6

**Comment by geoffturk** [comment]
It's looking like this might be an issue with the WordPress plug-in.

I scanned some works at random, and in that small sample all of them had "wp" directory links in them:
https://ipfs.io/ipfs/QmcpLPy9QPnQK2QwjcmpciWcN29S3xJaE6SwJTozKckETF
https://ipfs.io/ipfs/QmRsZuKJbJz2L9n7vXVZvstA2PggVn857oM2HwdbxQiLdg
https://ipfs.io/ipfs/Qma7Kin2F4Cx88DS7XFy8gs9vY2uZWspw2jnx36fTURjF6
https://ipfs.io/ipfs/QmUGNE7t1bUBC6EFCVmM5NcprsN24YnGPbfeUY4zdFfk59

Then I found this one from Crypto Insider in block 1290051:
https://ipfs.io/ipfs/QmRzcfahczChUDZKWWDhW4kqqvM1Ft5hMKepeBAnk8nz4s
Again, broken with "wp" directory links.

But then again from Crypto Insider a little later in block 1290058:
https://ipfs.io/ipfs/QmfTzdsJVc9Sy21xCjsrR6CJGWm94BfaqzzRBuEEV6iemt
Well-formed JSON and no "wp" directory links.

A bigger sample will be more conclusive but the WP plug-in seems to be the issue based on my research so far.

**Comment by lautarodragan** [comment]
The odd thing is that the node should reject invalid claims before they reach IPFS 🤔

**Comment by kennylavender** [comment]
@lautarodragan just throwing out a random guess because I am not familiar with streams. Could it be that the claim is valid, but because we are sending a stream the low timeout is cutting the stream off?

**Comment by kennylavender** [comment]
I was thinking the low timeout is probably fine for fast computers and servers, but if someone is running a node on a much slower computer, it could be timing out. Or they could be changing the configuration to point to a non-local ipfs server that requires more time.

**Comment by lautarodragan** [comment]
That could be it. How can we test / simulate a connection timing out? Maybe sending a lot of small files continuously for 30 seconds using our add function and a low timeout, or sending very large files...

**Comment by lautarodragan** [comment]
Note: if it was published by the WordPress Plugin then it came from Frost. Frost has 4 instances of the frost-api in production all hitting a single instance of the node. Maybe the single instance can't handle all the traffic at times? Or maybe the IPFS instance is the one unable to handle the traffic

**Comment by kennylavender** [comment]
> That could be it. How can we test / simulate a connection timing out? Maybe sending a lot of small files continuously for 30 seconds using our add function and a low timeout, or sending very large files...

I was thinking along the same lines for testing, large files and reducing the timeout until we start to see them timing out.

**Comment by geoffturk** [comment]
Reassigning to @kennylavender @lautarodragan
